lua/nvim-treesitter/ts_utils.lua

```diff
@@ -213,7 +213,7 @@ function M.memoize_by_buf_tick(fn)
       cache[bufnr] = {}
       api.nvim_buf_attach(bufnr, false,
         {
-          on_changedtick = function() cache[bufnr] = fn(bufnr) end,
```

Comment:
Not sure about this one, I think we want that to stay sync, since in some case we rely on the lazyness of this

Reply:
I put that in there because it makes the typing latency much lower and makes everything generally feel smoother, especially on files where tree sitter is slower like rust and markdown.

Reply:
Maybe I can remove that for now and put that in a separate pull request? The other async function seems fine and should improve performance a lot when opening a file. I have tested it on a rust file and it opened from ~700ms to ~300ms.